### PR TITLE
[Refactor](BE) Uniform Expr, Block, Columns size-relative interface (Part I)

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

In this pr we refactored some base interface to avoid narrow cast with data consistent kept.

We uniformed these type of numbers:
```
columns in Block        - U32 (just use no more than U16)
Rows in Block           - U64 (others U32, ColumnString64 U64)
column_id in expr arg   - I32 (because -1 is meaningful)
child of Expr           - U16 (FE use int32, have a restriction of 10000. adjustable but no guarantee for result)
```

So, for functions' argument column id, we use `uint32` to replace `size_t`.
For rows, we use `size_t` to replace some `int`.
For child of expr, we cast it as `uint_16` to use, rather than `size_t` as `vector::size()`

Refactor of table functions: https://github.com/apache/doris/pull/43218

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/exprs/vruntimefilter_wrapper.cpp [hunk 1]
```diff
@@ -99,7 +94,7 @@ Status VRuntimeFilterWrapper::execute(VExprContext* context, Block* block, int*
         size_t size = block->rows();
         block->insert({create_always_true_column(size, _data_type->is_nullable()), _data_type,
                        expr_name()});
-        *result_column_id = block->columns() - 1;
+        *result_column_id = static_cast<int>(block->columns() - 1);
```
Comment (Mryange) [inline be/src/vec/exprs/vruntimefilter_wrapper.cpp:97]: use cast set

File: be/src/vec/exprs/lambda_function/varray_map_function.cpp [hunk 2]
```diff
@@ -109,23 +105,23 @@ class ArrayMapFunction : public LambdaFunction {
             }
 
             // here is the array column
-            const ColumnArray& col_array = assert_cast<const ColumnArray&>(*column_array);
+            const auto& col_array = assert_cast<const ColumnArray&>(*column_array);
             const auto& col_type = assert_cast<const DataTypeArray&>(*type_array);
 
             if (i == 0) {
-                nested_array_column_rows = col_array.get_data_ptr()->size();
+                nested_array_column_rows = static_cast<int>(col_array.get_data_ptr()->size());
```
Comment (HappenLee) [inline be/src/vec/exprs/lambda_function/varray_map_function.cpp:112]: why here need cast？ not use `cast_set ` to do the work

File: be/src/vec/exprs/vruntimefilter_wrapper.h [hunk 3]
```diff
@@ -89,8 +87,8 @@ class VRuntimeFilterWrapper final : public VExpr {
         update_counters(filter_rows, input_rows);
 
         if (!_always_true) {
-            _judge_filter_rows += filter_rows;
-            _judge_input_rows += input_rows;
+            _judge_filter_rows += cast_set<int>(filter_rows);
+            _judge_input_rows += cast_set<int>(input_rows);
```
Comment (Mryange) [inline be/src/vec/exprs/vruntimefilter_wrapper.h:91]: 把 _judge_input_rows 换成atomic int 64吧

File: be/src/vec/exprs/table_function/vexplode_json_array.cpp [hunk 4]
```diff
@@ -144,7 +143,8 @@ void VExplodeJsonArrayTableFunction<DataImpl>::_insert_values_into_column(Mutabl
         size_t old_size = nullmap_column->size();
         nullmap_column->resize(old_size + max_step);
         memcpy(nullmap_column->get_data().data() + old_size,
-               _parsed_data.get_null_flag_address(_cur_offset), max_step * sizeof(UInt8));
+               _parsed_data.get_null_flag_address(cast_set<int>(_cur_offset)),
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_json_array.cpp:146]: why here need cast？ not use cast_set  to do the work where here really need int?

File: be/src/vec/exprs/table_function/vexplode_json_array.h [hunk 5]
```diff
@@ -77,7 +74,7 @@ struct ParsedDataInt : public ParsedData<int64_t> {
                 }
             } else if (v.IsDouble()) {
                 auto value = v.GetDouble();
-                if (value > MAX_VALUE) {
+                if (static_cast<long double>(value) > MAX_VALUE) {
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_json_array.h:77]: why double max use int64_t::max?

File: be/src/vec/exprs/vcompound_pred.h [hunk 6]
```diff
@@ -168,13 +170,13 @@ class VCompoundPred : public VectorizedFnCall {
         bool lhs_is_nullable = lhs_column->is_nullable();
         auto [lhs_data_column, lhs_null_map] =
                 _get_raw_data_and_null_map(lhs_column, lhs_is_nullable);
-        int filted = simd::count_zero_num((int8_t*)lhs_data_column, size);
+        int filted = static_cast<int>(simd::count_zero_num((int8_t*)lhs_data_column, size));
```
Comment (HappenLee) [inline be/src/vec/exprs/vcompound_pred.h:173]: really need int?

File: be/src/vec/exprs/vectorized_agg_fn.cpp [hunk 7]
```diff
@@ -355,7 +356,7 @@ Status AggFnEvaluator::check_agg_fn_output(int64_t key_size,
                                            const std::vector<vectorized::AggFnEvaluator*>& agg_fn,
                                            const RowDescriptor& output_row_desc) {
     auto name_and_types = VectorizedUtils::create_name_and_data_types(output_row_desc);
-    for (int i = key_size, j = 0; i < name_and_types.size(); i++, j++) {
+    for (int i = static_cast<int>(key_size), j = 0; i < name_and_types.size(); i++, j++) {
```
Comment (HappenLee) [inline be/src/vec/exprs/vectorized_agg_fn.cpp:359]: ?

File: be/src/vec/exprs/vexpr_context.cpp [hunk 8]
```diff
@@ -164,7 +168,7 @@ Status VExprContext::execute_conjuncts(const VExprContextSPtrs& ctxs,
                                        const std::vector<IColumn::Filter*>* filters,
                                        bool accept_null, Block* block,
                                        IColumn::Filter* result_filter, bool* can_filter_all) {
-    int rows = block->rows();
+    int rows = cast_set<int>(block->rows());
```
Comment (HappenLee) [inline be/src/vec/exprs/vexpr_context.cpp:171]: why here use int

File: be/src/vec/exprs/vruntimefilter_wrapper.h [hunk 9]
```diff
@@ -89,8 +87,8 @@ class VRuntimeFilterWrapper final : public VExpr {
         update_counters(filter_rows, input_rows);
 
         if (!_always_true) {
-            _judge_filter_rows += filter_rows;
-            _judge_input_rows += input_rows;
+            _judge_filter_rows += cast_set<int>(filter_rows);
```
Comment (HappenLee) [inline be/src/vec/exprs/vruntimefilter_wrapper.h:90]: int ?

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/exprs/vbitmap_predicate.cpp [hunk 10]
```diff
@@ -90,14 +87,14 @@ doris::Status vectorized::VBitmapPredicate::execute(vectorized::VExprContext* co
...
-    size_t num_columns_without_result = block->columns();
+    uint32_t num_columns_without_result = block->columns();
     auto res_data_column = ColumnVector<UInt8>::create(block->rows());
 
     ColumnPtr argument_column =
             block->get_by_position(arguments[0]).column->convert_to_full_column_if_const();
     size_t sz = argument_column->size();
     res_data_column->resize(sz);
-    auto ptr = ((ColumnVector<UInt8>*)res_data_column.get())->get_data().data();
+    auto* ptr = ((ColumnVector<UInt8>*)res_data_column.get())->get_data().data();
```
Comment (Mryange) [inline be/src/vec/exprs/vbitmap_predicate.cpp:97]: why type conversion here
```suggestion
    auto* ptr = ((ColumnVector<UInt8>*)res_data_column.get())->get_data().data();
```

File: be/src/vec/exprs/vexpr_context.cpp [hunk 11]
```diff
@@ -200,9 +204,10 @@ Status VExprContext::execute_conjuncts(const VExprContextSPtrs& ctxs,
                     }
                 }
 
-                int output_rows =
-                        rows -
-                        (is_rf_wrapper ? simd::count_zero_num((int8*)result_filter_data, rows) : 0);
+                size_t output_rows =
+                        rows - (is_rf_wrapper ? static_cast<int>(simd::count_zero_num(
```
Comment (Mryange) [inline be/src/vec/exprs/vexpr_context.cpp:208]: why static_cast

File: be/src/vec/exprs/vmatch_predicate.cpp [hunk 12]
```diff
@@ -181,14 +183,15 @@ const std::string& VMatchPredicate::function_name() const {
 std::string VMatchPredicate::debug_string() const {
     std::stringstream out;
     out << "MatchPredicate(" << children()[0]->debug_string() << ",[";
-    int num_children = children().size();
+    size_t num_children = get_num_children();
```
Comment (Mryange) [inline be/src/vec/exprs/vmatch_predicate.cpp:186]: uint16_t ?

Comment (zclllyybb) [reply be/src/vec/exprs/vmatch_predicate.cpp:186]: typo. thx~

**Comment by zclllyybb** [comment]
run buildall

File: be/src/util/simd/bits.h [hunk 13]
```diff
@@ -125,8 +125,10 @@ void iterate_through_bits_mask(Func func, decltype(bytes_mask_to_bits_mask(nullp
 #endif
 }
 
-inline size_t count_zero_num(const int8_t* __restrict data, size_t size) {
-    size_t num = 0;
+template <typename T>
```
Comment (HappenLee) [inline be/src/util/simd/bits.h:128]: maybe better add a default type ?

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Mryange — APPROVED** [review]

**Pull request merged**